arroyo-worker/src/connectors/mqtt/mod.rs

```diff
@@ -0,0 +1,131 @@
+use std::{sync::Arc, time::Duration};
+
+use anyhow::anyhow;
+use arroyo_rpc::var_str::VarStr;
+use rumqttc::v5::{AsyncClient, EventLoop, MqttOptions};
+use rustls_native_certs::load_native_certs;
+use serde::{Deserialize, Serialize};
+use tokio_rustls::rustls::{Certificate, ClientConfig, PrivateKey, RootCertStore};
+use typify::import_types;
+
+pub mod sink;
+pub mod source;
+
+import_types!(schema = "../connector-schemas/mqtt/connection.json",
+    convert = {
+        {type = "string", format = "var-str"} = VarStr
+    }
+);
+import_types!(schema = "../connector-schemas/mqtt/table.json");
+
+fn load_certs(certificates: &str) -> anyhow::Result<Vec<Certificate>> {
+    let cert_bytes = std::fs::read_to_string(certificates).map_or_else(
+        |_| certificates.as_bytes().to_owned(),
+        |certs| certs.as_bytes().to_owned(),
+    );
+
+    let certs = rustls_pemfile::certs(&mut cert_bytes.as_slice()).map_err(|err| anyhow!(err))?;
+
+    Ok(certs.into_iter().map(Certificate).collect())
+}
+
+fn load_private_key(certificate: &str) -> anyhow::Result<PrivateKey> {
+    let cert_bytes = std::fs::read_to_string(certificate).map_or_else(
+        |_| certificate.as_bytes().to_owned(),
+        |cert| cert.as_bytes().to_owned(),
+    );
+
+    let certs = rustls_pemfile::pkcs8_private_keys(&mut cert_bytes.as_slice())
+        .map_err(|err| anyhow!(err))?;
+    let cert = certs
+        .into_iter()
+        .next()
+        .ok_or_else(|| anyhow!("No private key found"))?;
+    Ok(PrivateKey(cert))
+}
+
+pub(crate) fn create_connection(
+    c: MqttConfig,
+    task_id: usize,
+) -> anyhow::Result<(AsyncClient, EventLoop)> {
+    let port = match c.port.map(|p| p as u16) {
+        Some(port) => port,
+        None => match c.protocol {
+            Protocol::Tcp => 1883,
+            Protocol::Tls => 8883,
+        },
+    };
+
+    let client_id = format!(
+        "{}_{}{}",
+        c.client_prefix
+            .as_ref()
+            .map(|s| s.as_str())
+            .unwrap_or_else(|| "arroyo-mqtt"),
+        task_id,
+        std::time::SystemTime::now()
+            .duration_since(std::time::UNIX_EPOCH)
+            .unwrap()
+            .as_millis()
+            % 100000,
```

Comment:
It creates a client id with the format: <client_prefix>_<task_id><current_time_in_millis> because the client id must be unique for each connection. Otherwise, the broker will only keep one active connection per client id